pkg/controller/masteruserrecord/masteruserrecord_controller.go

```diff
@@ -183,15 +181,20 @@ func (r *ReconcileMasterUserRecord) ensureUserAccount(logger logr.Logger, murAcc
 		recordSpecUserAcc: murAccount,
 		logger:            logger,
 	}
-	if err := sync.synchronizeSpec(); err != nil {
-		return r.wrapErrorWithStatusUpdate(logger, mur, r.setStatusFailed(toolchainv1alpha1.MasterUserRecordUnableToSynchronizeUserAccountSpecReason), err,
-			"update of the UserAccount.spec in the cluster '%s' failed", murAccount.TargetCluster)
+	if synchronized, err := sync.synchronizeSpec(); err != nil {
+		// if we got an error while synchronizing the specs, then we probably can't update it here neither.
+		// so we just return the error
+		return false, err
+	} else if synchronized {
+		return true, nil
```

Comment:
We still need to sync the status. We can't just drop here. Otherwise we won't change the MUR status when updating UserAccount spec during spec sync.
Imagine we have MUR and UserAccounts in sync with Ready status.
Then MUR spec is changed and we have just requested UserAccount update in line 184. The MUR status would still be "Ready" instead of updating it to Not Ready until the UserAccounts are now Ready.

Reply:
I also reverted this change

Reply:
I don't see this reverted. What am I missing?

Reply:
my bad, I reverted a change on some previous line and I responded too quickly on this one :|

Reply:
@alexeykazakov I reverted the change in 00a9739